[CI] need to add check_mode attr to TaskMock for synchronize

##### SUMMARY
Need to add the `check_mode` attribute to TaskMock class for the synchronizing plugin to address the following unit test issue:

- https://ansible.softwarefactory-project.io/zuul/build/4b6130d00ae34f0daf798403b480836a
- Addresses to #267 

```python
        if self._task.async_val and not self._supports_async:
            raise AnsibleActionFail('async is not supported for this task.')
>       elif self._task.check_mode and not self._supports_check_mode:
E       AttributeError: 'TaskMock' object has no attribute 'check_mode'

/tmp/ansible-test-sh44qo8i/ansible/plugins/action/__init__.py:106: AttributeError
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
- ansible.posix.tests.unit

##### ANSIBLE VERSION
- devel

##### COLLECTION VERSION
```paste below
main
```

##### CONFIGURATION
None

##### OS / ENVIRONMENT
None

##### STEPS TO REPRODUCE
```shell
$ ansible-test units --docker default -v
```
##### EXPECTED RESULTS
The unit tests are successful without error.

##### ACTUAL RESULTS

The unit tests have been failed with the following error:

```python
        if self._task.async_val and not self._supports_async:
            raise AnsibleActionFail('async is not supported for this task.')
>       elif self._task.check_mode and not self._supports_check_mode:
E       AttributeError: 'TaskMock' object has no attribute 'check_mode'

/tmp/ansible-test-sh44qo8i/ansible/plugins/action/__init__.py:106: AttributeError
```

--- a/changelogs/fragments/353_ci_fix_unittest_for_synchronize.yml
+++ b/changelogs/fragments/353_ci_fix_unittest_for_synchronize.yml
@@ -1,0 +1,3 @@
+---
+trivial:
+- CI tests - added check_mode attribute to TaskMock class for synchronize plugin unit test (https://github.com/ansible-collections/ansible.posix/issues/352).

--- a/tests/unit/plugins/action/test_synchronize.py
+++ b/tests/unit/plugins/action/test_synchronize.py
@@ -55,6 +55,7 @@ class TaskMock(object):
     become = None
     become_user = None
     become_method = None
+    check_mode = False
 
 
 class StdinMock(object):

